### ISSUE TITLE
How to impose additional validation logic?

### ISSUE THREAD

**Issue by nickjanssen** [microsoft/TypeChat#91]
This library is great, but how can I add additional validation? 
For example, how can I ensure that a string doesn't contain newlines?

GPT3 will often ignore instructions to return a string in a certain format, and it would be great if this library could retry until the right response (e.g. passing validation) was returned.

**Comment by ahejlsberg** [comment]
Yeah, we need the ability for user defined validation logic to execute and possibly convey explanatory diagnostics in the repair attempts. I will look at adding support for that.
